roles/backend/templates/backend.yml.j2

```diff
@@ -3,7 +3,9 @@
 # {{ managed }}
 #
 
+{% if not backend_config or not backend_config["state-dir"] %}
```

Comment:
Does this conditional survives this:

    vars:
      backend_config:
        debug: yes
    roles:
      - sensu.sensu_go.backend

I would expect the second part of the conditional to explode. If this is the case, maybe we should use `"state-dir" not in backend_config`?

Reply:
Thanks for spotting this issue. I have updated the code and also added another piece of tests.